### ISSUE TITLE
Rails 6 and Dependency issues (Thor)

### ISSUE THREAD

**Issue by joshm1204** [argerim/select2-rails#194]
Trying to upgrade to rails 6 and got the below error.

```

rails (~> 6.0, >= 6.0.2.1) was resolved to 6.0.2.1, which depends on
      railties (= 6.0.2.1) was resolved to 6.0.2.1, which depends on
        thor (>= 0.20.3, < 2.0)

    select2-rails was resolved to 4.0.3, which depends on
      thor (~> 0.14)

```

I assume select2-rails needs to update it's dependency requirements to make it compatible with rails 6.

**Comment by williamweckl** [comment]
I think you just have to update `thor`. Thor version 0.20.3 is compatible with 0.14. I know because we have thor version 0.20.3 installed and select2-rails.

Update your thor library:
`bundle update thor`

**Comment by seb-sykio** [comment]
still it would be wonderful to have an update with thor 1.0.1

**Comment by williamweckl** [comment]
@seb-sykio Would be perfect! Looking forward for the result of this thread: https://github.com/argerim/select2-rails/issues/193#issuecomment-571320673

**Comment by Vasfed** [comment]
Thor is only used for `rake update-select2` task that is used to update select2 code from upstream (also renames i18n files), I think it should be a development dependency, or at least optional

**Comment by ghost** [comment]
@Vasfed You are correct, for this gem thor is just for development. There is also a rake task that will update the select2 dependencies. Anyhow, select2-rails now requires thor `~> 1` hopefully that resolves this issue.

@joshm1204 Can you test this with your rails 6 app?
